### DISCUSSION TITLE
eCommerce integrations

### DISCUSSION THREAD

**Discussion by PD-Tony** [Budibase/budibase#642] [Ideas]
Hi there,

New here and currently booked into a onboarding session next month.

I work for an eCommerce company and looking to build a new internal business system to manage our online orders, production and various other bits.

What I really need is to be able to pull orders from our eCommerce platform (Magento 2) as well as push updates to orders.

Magento 2 has a REST API to allow for this but wondering what your timeline is for adding something like a generic REST API call/response feature?

Obviously this would be good not just for Magento 2 but for many other platforms that provide a REST API.

**Comment by shogunpurple** [comment]
Hi Tony,

Thanks very much for taking the time to write this up. We look forward to welcoming you onboard next month!

API integrations (specifically REST) are something we are currently working on as part of our “workflows” feature. We have created some workflow blocks that allow integration with zapier and Integromat through webhooks, so at the moment, you would be able to use one of those automation platforms to interact with magento and push/pull data from budibase.

I love the idea of a generic webhook block - this would be very useful to allow people to interact with any REST API. This is not something that is on our immediate roadmap for launch, but I can see it being something we will almost certainly need to build soon afterwards. 
 
All of our workflow blocks are open source and viewable at https://github.com/Budibase/automations. We allow contributions from the community here. If direct magento integration or a generic REST integration is something you might be interested in contributing to - feel free to open an issue over there.

What would make your ideal REST API integration feature in a no-code platform? Are there certain things you would like to able to configure when interacting with external APIs in a no code tool and how do you currently do this in your company?

**Reply by PD-Tony** [reply]
Hi Martin,

Thanks for your response.

Great to hear Integromat is supported, I'll have a deeper look into that as I know that supports Magento 2 as well as other integrations.

I too thought a generic webhook block could be useful, it will then reduce the need to write specific hooks for everything and allow the consumers to talk to almost any API. I would expect it to be a bit of a pain to figure out a UI to make this easy for users but definitely possible.

I'll have a look into the automations github once I've been on-boarded and played with Budibase and gotten familiar with it. I'm a web developer with about 20 years experience now, so I'd be happy to contribute bits of code that I need Budibase to support ;-).

Not sure on what my ideal integration feature for a REST API would be at this time, but I'll have a think and come back to you.

Looking forward to getting my hands on Budibase to develop a replace for our current in-house business system. I do have the skills to code this by hand but I lack the time with other tasks and need something like Budibase to build rapidly, test and iterate - so seems a good fit.

**Reply by shogunpurple** [reply]
That's fantastic! We are always happy to have more developers in the budibase community with ideas and contributions to our platform - it's one of the reasons we fully embrace the open source nature of our product.

I'd love to chat some more once you've been onboarded. We are always all ears when it comes to product requests, especially for automations. Feel free to continue the discussion here if anything comes to mind around your ideal REST API integration feature. I'll be sure to keep you in the loop if we do decide to add it to our roadmap or implement it any time soon.

**Reply by PD-Tony** [reply]
Thanks for the warm welcome and I look forward to using Budibase and contributing thoughts, ideas and more as they come to me (I'm sure I'll have loads).

Happy to have a further chat post onboarding, I can talk you through the project that I'll be using Budibase to develop and you can give me your thoughts.

**Comment by mapmychannel** [comment]
Hi all,

I’m starting this thread to dive into [eCommerce integrations](https://www.mapmychannel.com/blog/ecommerce-integration-what-businesses-need-to-know-in-2025)—specifically, which ones are truly essential for improving performance, automating workflows, and scaling efficiently in today’s landscape.

With so many tools and APIs on the market, the challenge isn’t just choosing the right platform (Shopify, WooCommerce, Magento, BigCommerce, etc.)—it’s about selecting the right integrations that solve real operational bottlenecks and actually drive value.

I’d love to hear what’s working (and what’s not) from both a business and technical perspective.
